src/main/java/org/prebid/server/bidder/teads/TeadsBidder.java

```diff
@@ -0,0 +1,164 @@
+                .map(format -> banner.toBuilder().w(format.getW()).h(format.getH()).build())
+                .orElse(banner)
+                : null;
+    }
+
+    private TeadsImpExt parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest bidRequest) {
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .impIds(BidderUtil.impIds(bidRequest))
+                .body(mapper.encodeToBytes(bidRequest))
+                .payload(bidRequest)
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.withValues(extractBids(httpCall.getRequest().getPayload(), bidResponse));
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse("Bad Server Response"));
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Empty SeatBid array");
+        }
+
+        final Map<String, Imp> impMap = bidRequest.getImp().stream()
+                .collect(Collectors.toMap(Imp::getId, Function.identity()));
+
+        return bidResponse.getSeatbid()
+                .stream()
+                .flatMap(seatBid -> Optional.ofNullable(seatBid.getBid()).orElse(List.of()).stream())
+                .map(bid -> makeBid(impMap, bid, bidResponse.getCur()))
+                .toList();
+    }
+
+    private BidderBid makeBid(Map<String, Imp> impMap, Bid bid, String currency) {
+        final BidType bidType = Optional.ofNullable(impMap.get(bid.getImpid()))
+                .map(imp -> imp.getVideo() == null ? BidType.banner : BidType.video)
+                .orElseThrow(() -> new PreBidException("Bid for the Imp " + bid.getImpid() + " wasn't found"));
+
+        final ExtBidPrebidMeta meta = parseExtBidPrebidMeta(bid);
+        if (StringUtils.isBlank(meta.getRendererName())) {
+            throw new PreBidException("RendererName should not be empty");
+        }
+
+        if (StringUtils.isBlank(meta.getRendererVersion())) {
+            throw new PreBidException("RendererVersion should not be empty");
+        }
+
+        return BidderBid.of(bid, bidType, currency);
```

Comment:
Looks like In Go they completely creates new meta-object, with just two fields

Reply:
that's not true due to the test cases they cover, please check some of them

Reply:
// TypedBid.BidMeta will become "response.seatbid[i].bid.ext.prebid.meta" in the final OpenRTB response

They have this data only in bid, but in final version they will have only this two fields